packages/utils/src/feature_toggle.rs

```diff
@@ -0,0 +1,393 @@
+};
+use cosmwasm_storage::{Bucket, ReadonlyBucket};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::fmt::Display;
+
+const PREFIX_FEATURES: &[u8] = b"features";
+const PREFIX_PAUSERS: &[u8] = b"pausers";
+
+pub struct FeatureToggle;
+
+impl FeatureToggleStore for FeatureToggle {
+    const STORAGE_KEY: &'static [u8] = b"feature_toggle";
+}
+
+pub trait FeatureToggleStore {
+    const STORAGE_KEY: &'static [u8];
+
+    fn init_features<S: Storage, T: Serialize>(
+        storage: &mut S,
+        feature_statuses: Vec<FeatureStatus<T>>,
+        pausers: Vec<HumanAddr>,
+    ) -> StdResult<()> {
+        for feature_status in feature_statuses {
+            Self::set_feature_status(storage, feature_status.feature, feature_status.status)?;
+        }
+
+        for p in pausers {
+            Self::set_pauser(storage, p)?;
+        }
+
+        Ok(())
+    }
+
+    fn pause<S: Storage, T: Serialize>(storage: &mut S, features: Vec<T>) -> StdResult<()> {
+        for f in features {
+            Self::set_feature_status(storage, f, Status::Paused)?;
+        }
+
+        Ok(())
+    }
+
+    fn unpause<S: Storage, T: Serialize>(storage: &mut S, features: Vec<T>) -> StdResult<()> {
+        for f in features {
+            Self::set_feature_status(storage, f, Status::NotPaused)?;
+        }
+
+        Ok(())
+    }
+
+    fn get_pauser<S: ReadonlyStorage>(storage: &S, key: HumanAddr) -> StdResult<Option<u8>> {
+        let feature_store =
+            ReadonlyBucket::multilevel(&[Self::STORAGE_KEY, PREFIX_PAUSERS], storage);
+        feature_store.may_load(key.0.as_bytes())
+    }
+
+    fn set_pauser<S: Storage>(storage: &mut S, key: HumanAddr) -> StdResult<()> {
+        let mut feature_store = Bucket::multilevel(&[Self::STORAGE_KEY, PREFIX_PAUSERS], storage);
+        feature_store.save(key.0.as_bytes(), &1_u8 /* value is insignificant */)
+    }
+
+    fn remove_pauser<S: Storage>(storage: &mut S, key: HumanAddr) {
+        let mut feature_store: Bucket<S, u8> =
+            Bucket::multilevel(&[Self::STORAGE_KEY, PREFIX_PAUSERS], storage);
+        feature_store.remove(key.0.as_bytes())
+    }
+
+    fn get_feature_status<S: ReadonlyStorage, T: Serialize>(
+        storage: &S,
+        key: T,
+    ) -> StdResult<Option<Status>> {
+        let feature_store =
+            ReadonlyBucket::multilevel(&[Self::STORAGE_KEY, PREFIX_FEATURES], storage);
+        feature_store.may_load(&cosmwasm_std::to_vec(&key)?)
+    }
+
+    fn set_feature_status<S: Storage, T: Serialize>(
+        storage: &mut S,
+        key: T,
+        item: Status,
+    ) -> StdResult<()> {
+        let mut feature_store = Bucket::multilevel(&[Self::STORAGE_KEY, PREFIX_FEATURES], storage);
+        feature_store.save(&cosmwasm_std::to_vec(&key)?, &item)
+    }
+}
+
+impl FeatureToggle {
+    pub fn require_not_paused<S: Storage, T: Serialize + Display + Clone>(
+        storage: &S,
+        features: Vec<T>,
+    ) -> StdResult<()> {
```

Comment:
I think that all the methods should be under the trait. Otherwise if someone overrides the functionality of `FeatureToggle`, they have to copy-paste these methods.

Reply:
You're right 👍🏼